app.rb

```diff
@@ -1,7 +1,27 @@
 require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/play'
+
 class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+
+  enable :sessions
+
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  post '/name' do
+    session[:player1_name] = params[:player1_name]
+    redirect '/play'
+  end
+
+  get '/play' do
+    @player1_name = session[:player1_name]
```

Comment:
I found you don't need this line at all. 

The whole point in sessions Is they are permanent variables throughout your work, so you just call session[:player1_name] wherever you want to use it, instead of calling it @player1_name and calling that.

I know they did that in the walkthrough, but I have absolutely no idea why.